### PR TITLE
Add locale value into pages link 

### PR DESCRIPTION
**Pull request by BaseMax**
Hi Folks,

The current version of the website is not good and has some problems.
If you want to go to other pages, the language becomes English again.
And it does not make good sense to readers.

Regards,

### CODE REVIEW

**Review by MikeMcQuaid — COMMENTED** [review]
> The current version of the website is not good and has some problems.
> If you want to go to other pages, the language becomes English again.
> And it does not make good sense to readers.

I cannot reproduce that. Can you explain your workflow? If I change my language and click links: they load as the language I had previously selected.

**Comment by BaseMax** [comment]
Hi Mike,

> Can you explain your workflow?

Yes, sure.

> They load as the language I had previously selected.

Not, Did not.

See source code. It does not transmit `locale` value.
And if the locale parameter does not exist, the default will be English.

1- If you choose **FR** language.
2- Then go to another section from the top menu. It will be **English** again.
The selected language must be preserved.

This seems clear.
[video-demo.zip](https://github.com/github/opensourcefriday/files/5574618/video-demo.zip)

**Comment by MikeMcQuaid** [comment]
@BaseMax can you still reproduce this if you sign n?

**Comment by BaseMax** [comment]
This works when I sign in. :)

**Comment by MikeMcQuaid** [comment]
@BaseMax Sorry for the delay! Given that: could this logic be made conditional on users being signed-out? Creating a new e.g. helper for that may be useful to avoid duplication, too. Thanks!

**Comment by BaseMax** [comment]
Hi Mike,

Thank you for your answer.

Review this if possible. @MikeMcQuaid 
Currently we only transfer the locale value when the user is a guest.

Please comment. What do you think?
Is it a good solution?

File: app/views/layouts/_nav.html.erb [hunk 1]
```diff
@@ -1,16 +1,28 @@
 <nav class="text-center">
-  <a href="/#participate"><%=t "Contributors" %></a>
-
+  <% if current_user %>
+    <a href="/#participate"><%=t "Contributors" %></a>
+  <% else %>
+    <a href="/<%= "?locale=#{I18n.locale}" if I18n.locale != :en %>#participate"><%=t "Contributors" %></a>
```
Comment (MikeMcQuaid) [inline app/views/layouts/_nav.html.erb:5]:
```suggestion
    <a href="/<%= "?locale=#{I18n.locale}" if I18n.locale != :en %>#participate"><%=t "Contributors" %></a>
```

I'd like to see the i18n logic and `current_user` logic pulled into a (new) helper as it's repeated three times.

Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: app/views/layouts/_nav.html.erb (hunk 1 above)
Comment (MikeMcQuaid) [reply app/views/layouts/_nav.html.erb:5]: @BaseMax nudge on this when you get a chance!

Comment (BaseMax) [reply app/views/layouts/_nav.html.erb:5]: Hi Mike, Sure I will test this solution as soon as possible. Thanks @MikeMcQuaid

**Comment by MikeMcQuaid** [comment]
Passing on this as it seems to have stalled, sorry!

**Pull request closed without merging**